core/src/main/java/com/turn/ttorrent/common/Torrent.java

```diff
@@ -711,21 +715,22 @@ public String call() throws UnsupportedEncodingException {
 	 * @param file The file to hash.
 	 */
 	private static String hashFile(File file, int pieceLenght)
-		throws InterruptedException, IOException, NoSuchAlgorithmException {
+		throws InterruptedException, IOException, NoSuchAlgorithmException, Exception {
 		return Torrent.hashFiles(Arrays.asList(new File[] { file }), pieceLenght);
 	}
 
 	private static String hashFiles(List<File> files, int pieceLenght)
-		throws InterruptedException, IOException, NoSuchAlgorithmException {
+		throws InterruptedException, IOException, NoSuchAlgorithmException, Exception {
 		int threads = getHashingThreadsCount();
 		ExecutorService executor = Executors.newFixedThreadPool(threads);
-		ByteBuffer buffer = ByteBuffer.allocate(pieceLenght);
+		final ByteBufferPool bbp = new ByteBufferPool(threads + 1, pieceLenght);
```

Comment:
haha, ok, didn't even notice, though it will generate noise around the patch itself, since the typos are unrelated